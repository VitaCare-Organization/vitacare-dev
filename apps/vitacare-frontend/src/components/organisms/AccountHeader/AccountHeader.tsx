export default function AccountHeader() {
    return (
      <div className="text-center space-y-2 mb-6" data-testid="account-header">
        <div className="flex justify-center">
          <div className="w-10 h-10 rounded-full bg-blue-50 flex items-center justify-center">
            <svg
              width="24"
              height="24"
              viewBox="0 0 24 24"
              fill="none"
              xmlns="http://www.w3.org/2000/svg"
              className="text-primary"
            >
              <path
                d="M12 12C14.7614 12 17 9.76142 17 7C17 4.23858 14.7614 2 12 2C9.23858 2 7 4.23858 7 7C7 9.76142 9.23858 12 12 12Z"
                stroke="currentColor"
                strokeWidth="2"
                strokeLinecap="round"
                strokeLinejoin="round"
              />
              <path
                d="M20 21C20 18.7909 18.7909 16.7909 17 15.5C15.2091 14.2091 13.2091 13.5 12 13.5C10.7909 13.5 8.79086 14.2091 7 15.5C5.20914 16.7909 4 18.7909 4 21"
                stroke="currentColor"
                strokeWidth="2"
                strokeLinecap="round"
                strokeLinejoin="round"
              />
            </svg>
          </div>
        </div>
        <h1 className="text-2xl font-bold text-gray-900">Create an account</h1>
        <p className="text-sm text-gray-500">Register to access the VitaCare platform</p>
      </div>
    )
  }
  
  