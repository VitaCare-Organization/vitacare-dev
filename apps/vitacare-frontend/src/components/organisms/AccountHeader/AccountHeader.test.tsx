import { render, screen } from "@testing-library/react";
import AccountHeader from "./AccountHeader";

describe("AccountHeader Component", () => {
  it("renders correctly with all elements", () => {
    render(<AccountHeader />);
    
    // Check for the main heading
    expect(screen.getByRole("heading", { level: 1 })).toHaveTextContent("Create an account");
    
    // Check for the description text
    expect(screen.getByText("Register to access the VitaCare platform")).toBeInTheDocument();
  });

  it("displays the user icon correctly", () => {
    render(<AccountHeader />);
    
    // Check for the icon container
    const iconContainer = screen.getByRole("heading", { level: 1 }).closest("div")?.querySelector("div > div");
    expect(iconContainer).toHaveClass("w-10 h-10 rounded-full bg-blue-50 flex items-center justify-center");
    
    // Check for SVG presence
    const svg = screen.getByRole("heading", { level: 1 }).closest("div")?.querySelector("svg");
    expect(svg).toBeInTheDocument();
    expect(svg).toHaveAttribute("width", "24");
    expect(svg).toHaveAttribute("height", "24");
  });

  it("has correct styling classes", () => {
    render(<AccountHeader />);
    
    const container = screen.getByRole("heading", { level: 1 }).closest("div");
    expect(container).toHaveClass("text-center space-y-2 mb-6");
    
    const heading = screen.getByRole("heading", { level: 1 });
    expect(heading).toHaveClass("text-2xl font-bold text-gray-900");
    
    const description = screen.getByText("Register to access the VitaCare platform");
    expect(description).toHaveClass("text-sm text-gray-500");
  });

  it("renders with proper accessibility", () => {
    render(<AccountHeader />);
    
    // Check that the heading is properly structured
    const heading = screen.getByRole("heading", { level: 1 });
    expect(heading).toBeInTheDocument();
    
    // Check that the description is visible to screen readers
    const description = screen.getByText("Register to access the VitaCare platform");
    expect(description).toBeVisible();
  });
}); 