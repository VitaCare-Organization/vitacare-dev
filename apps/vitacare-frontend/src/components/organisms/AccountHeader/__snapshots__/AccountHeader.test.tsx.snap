// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountHeader Component matches snapshot 1`] = `
<div
  class="text-center space-y-2 mb-6"
>
  <div
    class="flex justify-center"
  >
    <div
      class="w-10 h-10 rounded-full bg-blue-50 flex items-center justify-center"
    >
      <svg
        class="text-primary"
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 12C14.7614 12 17 9.76142 17 7C17 4.23858 14.7614 2 12 2C9.23858 2 7 4.23858 7 7C7 9.76142 9.23858 12 12 12Z"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
        <path
          d="M20 21C20 18.7909 18.7909 16.7909 17 15.5C15.2091 14.2091 13.2091 13.5 12 13.5C10.7909 13.5 8.79086 14.2091 7 15.5C5.20914 16.7909 4 18.7909 4 21"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </div>
  </div>
  <h1
    class="text-2xl font-bold text-gray-900"
  >
    Create an account
  </h1>
  <p
    class="text-sm text-gray-500"
  >
    Register to access the VitaCare platform
  </p>
</div>
`;
